src/OrchardCore/OrchardCore.DisplayManagement/Implementation/DisplayHelper.cs

```diff
@@ -75,7 +76,19 @@ public Task<IHtmlContent> ShapeExecuteAsync(IShape shape)
             // Check if the shape is pre-rendered.
             if (shape is IHtmlContent htmlContent)
             {
-                return Task.FromResult(htmlContent);
+                if (shape is PositionWrapper wrapper)
+                {
+                    while(wrapper.Value is PositionWrapper)
+                    {
+                        wrapper = (PositionWrapper) wrapper.Value;
+                    }
+                    
+                    return Task.FromResult(wrapper.Value);
+                }
+                else
+                {
```

Comment:
same here drop the use of else block and just return

Reply:
Removed